### PR TITLE
Add vulnerability scanner

### PR DESCRIPTION
**Pull request by strickvl**
Added 

- [Trivy](https://trivy.dev/) scans of the `latest` tags of our two Docker images (running whenever we merge to `main`, i.e. after release, and also once a week on Monday morning)
- [CodeQL scans](https://codeql.github.com/)

All this populates the [Code Scanning](https://github.com/zenml-io/zenml/security/code-scanning) section of our Security tab on this repo. We can then deal with any issues that come up as and when necessary.

## Pre-requisites
Please ensure you have done the following:
- [ ] I have read the **CONTRIBUTING.md** document.
- [ ] If my change requires a change to docs, I have updated the documentation accordingly.
- [ ] If I have added an integration, I have updated the [integrations](https://docs.zenml.io/stacks-and-components/component-guide) table and the [corresponding website section](https://zenml.io/integrations).
- [ ] I have added tests to cover my changes.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Other (add details above)

### CODE REVIEW

**Comment by fa9r** [comment]
> CodeQL scans ??? ;)

File: .github/workflows/codeql.yml [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+  analyze:
+    name: Analyze
+    # Runner size impacts CodeQL analysis time. To learn more, please see:
+    #   - https://gh.io/recommended-hardware-resources-for-running-codeql
+    #   - https://gh.io/supported-runners-and-hardware-resources
+    #   - https://gh.io/using-larger-runners
+    # Consider using larger runners for possible analysis time improvements.
+    runs-on:
+      ${{ (matrix.language == 'swift' && 'macos-latest') || 'ubuntu-latest' }}
+    timeout-minutes: ${{ (matrix.language == 'swift' && 120) || 360 }}
```
Comment (avishniakov) [inline .github/workflows/codeql.yml:21]: Do we need `swift` part here?

**Review by avishniakov — APPROVED** [review]
Good one, just nitpicking

**Review by fa9r — APPROVED** [review]
Nice, LGTM. 

If you have a bit more time you could perhaps clean up the CodeQL file a bit by removing all the unused / commented-out code. Otherwise no suggestions from my side.

One thing I didn't fully get is how CodeQL would be used in practice. Will that post a message on the PR or how does this work exactly?

**Comment by strickvl** [comment]
@fa9r answer to your question of how it works is in the PR description.

**Pull request merged**